src/util/Views.h

```diff
@@ -199,6 +200,31 @@ auto inPlaceTransformView(Range&& range, Transformation transformation) {
                                           std::move(transformation));
 }
 
+/// Create a generator the consumes the input generator until it finds a newline
+/// character and the yields views of whole lines.
+inline cppcoro::generator<std::string_view> lineByLine(
+    cppcoro::generator<std::span<std::byte>> generator) {
```

Comment:
This should take a templated input_range the reference type of which can be converted to `std::string_view`. No need to hardcode the generator requirement on the input,
And also the `std::byte` is odd. If that is a usage artifact of your HTTP buffers, then those need to handle it.